nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviService.java

```diff
@@ -45,134 +35,113 @@ public static NviService defaultNviService() {
         return new NviService(defaultDynamoClient());
     }
 
-    //TODO: Remove JacocoGenerated when other if/else cases are implemented
-    @JacocoGenerated
-    public Optional<CandidateWithIdentifier> upsertCandidate(CandidateEvaluatedMessage evaluatedCandidate) {
-        if (!isExistingCandidate(evaluatedCandidate) && isNviCandidate(evaluatedCandidate)) {
-            return Optional.of(createCandidate(evaluatedCandidate));
-        } else if (isExistingCandidate(evaluatedCandidate) && isNviCandidate(evaluatedCandidate)) {
-            var existing = findByPublicationId(evaluatedCandidate.candidateDetails().publicationId()).orElseThrow();
+    @JacocoGenerated //TODO Temporary for coverage
+    public Optional<Candidate> upsertCandidate(boolean isNviCandidate, DbCandidate candidate) {
+        if (!isExistingCandidate(candidate.publicationId()) && isNviCandidate) {
+            return Optional.of(
+                createCandidate(candidate,
+                                generatePendingApprovalStatuses(candidate.points())
+                ));
```

Comment:
If we pass DbCandidate only to this method, than we can call something like !isExistingAndApplicable()